src/SipiImage.cpp

```diff
@@ -508,68 +508,129 @@ namespace Sipi {
     /*==========================================================================*/
 
 
-    void SipiImage::removeChan(unsigned int chan) {
-        if ((nc == 1) || (chan >= nc)) {
-            std::string msg = "Cannot remove component: nc=" + std::to_string(nc) + " chan=" + std::to_string(chan);
+    void SipiImage::removeChannel(const unsigned int channel, const bool force_gray_alpha) {
+        if ((nc == 1) || (channel >= nc)) {
+            std::string msg = "Cannot remove component: nc=" + std::to_string(nc) + " chan=" + std::to_string(channel);
             throw SipiImageError(__file__, __LINE__, msg);
         }
 
-        if (!es.empty()) {
-            if (nc < 3) {
-                es.clear(); // no more alpha channel
-            } else if (nc > 3) { // it's probably an alpha channel
-                if ((nc == 4) && (photo == SEPARATED)) {  // oh no – 4 channels, but CMYK
-                    std::string msg =
-                            "Cannot remove component: nc=" + std::to_string(nc) + " chan=" + std::to_string(chan);
-                    throw SipiImageError(__file__, __LINE__, msg);
-                } else {
-                    es.erase(es.begin() + (chan - ((photo == SEPARATED) ? 4 : 3)));
-                }
-            } else {
-                std::string msg = "Cannot remove component: nc=" + std::to_string(nc) + " chan=" + std::to_string(chan);
+        const bool has_removable_extra_samples = !es.empty();
+        const bool has_two_or_less_channels = nc < 3;
+        const bool has_three_channels = nc == 3;
+
+        // Assumtion: An image with two or less channels cannot have extra samples
+        assert(has_two_or_less_channels && has_removable_extra_samples);
+
+        if (has_removable_extra_samples) {
+
+            // cleanup the extra samples
+            // TODO: figure out why this can even happen
```

Comment:
It is a reminder to my future self. The only reason why it is a todo is that clion colors it differently so that I can easily spot it. Ideally there should be no TODOs but only actionable issues on Linear. Unfortunately, at this point, I'm still figuring out what Sipi is doing. As soon as I know, it will be turned into an issue.